In [10]:
#Importing Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

In [8]:
# loading the data
data_features = pd.read_csv('training_set_features.csv') # predictors
data_target = pd.read_csv('training_set_labels.csv') # target

In [21]:
data_features.columns

Index(['respondent_id', 'h1n1_concern', 'h1n1_knowledge',
       'behavioral_antiviral_meds', 'behavioral_avoidance',
       'behavioral_face_mask', 'behavioral_wash_hands',
       'behavioral_large_gatherings', 'behavioral_outside_home',
       'behavioral_touch_face', 'doctor_recc_h1n1', 'doctor_recc_seasonal',
       'chronic_med_condition', 'child_under_6_months', 'health_worker',
       'health_insurance', 'opinion_h1n1_vacc_effective', 'opinion_h1n1_risk',
       'opinion_h1n1_sick_from_vacc', 'opinion_seas_vacc_effective',
       'opinion_seas_risk', 'opinion_seas_sick_from_vacc', 'age_group',
       'education', 'race', 'sex', 'income_poverty', 'marital_status',
       'rent_or_own', 'employment_status', 'hhs_geo_region', 'census_msa',
       'household_adults', 'household_children', 'employment_industry',
       'employment_occupation'],
      dtype='object')

In [22]:
data_target.columns

Index(['respondent_id', 'h1n1_vaccine', 'seasonal_vaccine'], dtype='object')

In [25]:
# Combining the two datasets
df = pd.concat([data_features, data_target], axis=1) # combining target and predictor data

In [20]:
df.head()

respondent_id  h1n1_concern  h1n1_knowledge  behavioral_antiviral_meds  \
0              0           1.0             0.0                        0.0   
1              1           3.0             2.0                        0.0   
2              2           1.0             1.0                        0.0   
3              3           1.0             1.0                        0.0   
4              4           2.0             1.0                        0.0   

   behavioral_avoidance  behavioral_face_mask  behavioral_wash_hands  \
0                   0.0                   0.0                    0.0   
1                   1.0                   0.0                    1.0   
2                   1.0                   0.0                    0.0   
3                   1.0                   0.0                    1.0   
4                   1.0                   0.0                    1.0   

   behavioral_large_gatherings  behavioral_outside_home  \
0                          0.0                      1.0   
1                          0.0                      1.0   
2                          0.0                      0.0   
3                          1.0                      0.0   
4                          1.0                      0.0   

   behavioral_touch_face  ...   employment_status  hhs_geo_region  \
0                    1.0  ...  Not in Labor Force        oxchjgsf   
1                    1.0  ...            Employed        bhuqouqj   
2                    0.0  ...            Employed        qufhixun   
3                    0.0  ...  Not in Labor Force        lrircsnp   
4                    1.0  ...            Employed        qufhixun   

                 census_msa  household_adults  household_children  \
0                   Non-MSA               0.0                 0.0   
1  MSA, Not Principle  City               0.0                 0.0   
2  MSA, Not Principle  City               2.0                 0.0   
3       MSA, Principle City               0.0                 0.0   
4  MSA, Not Principle  City               1.0                 0.0   

   employment_industry  employment_occupation  respondent_id  h1n1_vaccine  \
0                  NaN                    NaN              0             0   
1             pxcmvdjn               xgwztkwe              1             0   
2             rucpziij               xtkaffoo              2             0   
3                  NaN                    NaN              3             0   
4             wxleyezf               emcorrxb              4             0   

   seasonal_vaccine  
0                 0  
1                 1  
2                 0  
3                 1  
4                 0  

[5 rows x 39 columns]

In [11]:
df.columns

Index(['respondent_id', 'h1n1_concern', 'h1n1_knowledge',
       'behavioral_antiviral_meds', 'behavioral_avoidance',
       'behavioral_face_mask', 'behavioral_wash_hands',
       'behavioral_large_gatherings', 'behavioral_outside_home',
       'behavioral_touch_face', 'doctor_recc_h1n1', 'doctor_recc_seasonal',
       'chronic_med_condition', 'child_under_6_months', 'health_worker',
       'health_insurance', 'opinion_h1n1_vacc_effective', 'opinion_h1n1_risk',
       'opinion_h1n1_sick_from_vacc', 'opinion_seas_vacc_effective',
       'opinion_seas_risk', 'opinion_seas_sick_from_vacc', 'age_group',
       'education', 'race', 'sex', 'income_poverty', 'marital_status',
       'rent_or_own', 'employment_status', 'hhs_geo_region', 'census_msa',
       'household_adults', 'household_children', 'employment_industry',
       'employment_occupation', 'respondent_id', 'h1n1_vaccine',
       'seasonal_vaccine'],
      dtype='object')

In [12]:
df.shape

(26707, 39)

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26707 entries, 0 to 26706
Data columns (total 39 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   respondent_id                26707 non-null  int64  
 1   h1n1_concern                 26615 non-null  float64
 2   h1n1_knowledge               26591 non-null  float64
 3   behavioral_antiviral_meds    26636 non-null  float64
 4   behavioral_avoidance         26499 non-null  float64
 5   behavioral_face_mask         26688 non-null  float64
 6   behavioral_wash_hands        26665 non-null  float64
 7   behavioral_large_gatherings  26620 non-null  float64
 8   behavioral_outside_home      26625 non-null  float64
 9   behavioral_touch_face        26579 non-null  float64
 10  doctor_recc_h1n1             24547 non-null  float64
 11  doctor_recc_seasonal         24547 non-null  float64
 12  chronic_med_condition        25736 non-null  float64
 13  child_under_6_mo

In [18]:
# showing a transposed DataFrame where descriptive statistics are arranged by columns 
df.describe().T

count          mean          std  min     25%  \
respondent_id                26707.0  13353.000000  7709.791156  0.0  6676.5   
h1n1_concern                 26615.0      1.618486     0.910311  0.0     1.0   
h1n1_knowledge               26591.0      1.262532     0.618149  0.0     1.0   
behavioral_antiviral_meds    26636.0      0.048844     0.215545  0.0     0.0   
behavioral_avoidance         26499.0      0.725612     0.446214  0.0     0.0   
behavioral_face_mask         26688.0      0.068982     0.253429  0.0     0.0   
behavioral_wash_hands        26665.0      0.825614     0.379448  0.0     1.0   
behavioral_large_gatherings  26620.0      0.358640     0.479610  0.0     0.0   
behavioral_outside_home      26625.0      0.337315     0.472802  0.0     0.0   
behavioral_touch_face        26579.0      0.677264     0.467531  0.0     0.0   
doctor_recc_h1n1             24547.0      0.220312     0.414466  0.0     0.0   
doctor_recc_seasonal         24547.0      0.329735     0.470126  0.0     0.0   
chronic_med_condition        25736.0      0.283261     0.450591  0.0     0.0   
child_under_6_months         25887.0      0.082590     0.275266  0.0     0.0   
health_worker                25903.0      0.111918     0.315271  0.0     0.0   
health_insurance             14433.0      0.879720     0.325300  0.0     1.0   
opinion_h1n1_vacc_effective  26316.0      3.850623     1.007436  1.0     3.0   
opinion_h1n1_risk            26319.0      2.342566     1.285539  1.0     1.0   
opinion_h1n1_sick_from_vacc  26312.0      2.357670     1.362766  1.0     1.0   
opinion_seas_vacc_effective  26245.0      4.025986     1.086565  1.0     4.0   
opinion_seas_risk            26193.0      2.719162     1.385055  1.0     2.0   
opinion_seas_sick_from_vacc  26170.0      2.118112     1.332950  1.0     1.0   
household_adults             26458.0      0.886499     0.753422  0.0     0.0   
household_children           26458.0      0.534583     0.928173  0.0     0.0   
respondent_id                26707.0  13353.000000  7709.791156  0.0  6676.5   
h1n1_vaccine                 26707.0      0.212454     0.409052  0.0     0.0   
seasonal_vaccine             26707.0      0.465608     0.498825  0.0     0.0   

                                 50%      75%      max  
respondent_id                13353.0  20029.5  26706.0  
h1n1_concern                     2.0      2.0      3.0  
h1n1_knowledge                   1.0      2.0      2.0  
behavioral_antiviral_meds        0.0      0.0      1.0  
behavioral_avoidance             1.0      1.0      1.0  
behavioral_face_mask             0.0      0.0      1.0  
behavioral_wash_hands            1.0      1.0      1.0  
behavioral_large_gatherings      0.0      1.0      1.0  
behavioral_outside_home          0.0      1.0      1.0  
behavioral_touch_face            1.0      1.0      1.0  
doctor_recc_h1n1                 0.0      0.0      1.0  
doctor_recc_seasonal             0.0      1.0      1.0  
chronic_med_condition            0.0      1.0      1.0  
child_under_6_months             0.0      0.0      1.0  
health_worker                    0.0      0.0      1.0  
health_insurance                 1.0      1.0      1.0  
opinion_h1n1_vacc_effective      4.0      5.0      5.0  
opinion_h1n1_risk                2.0      4.0      5.0  
opinion_h1n1_sick_from_vacc      2.0      4.0      5.0  
opinion_seas_vacc_effective      4.0      5.0      5.0  
opinion_seas_risk                2.0      4.0      5.0  
opinion_seas_sick_from_vacc      2.0      4.0      5.0  
household_adults                 1.0      1.0      3.0  
household_children               0.0      1.0      3.0  
respondent_id                13353.0  20029.5  26706.0  
h1n1_vaccine                     0.0      0.0      1.0  
seasonal_vaccine                 0.0      1.0      1.0

In [14]:
df.dtypes

respondent_id                    int64
h1n1_concern                   float64
h1n1_knowledge                 float64
behavioral_antiviral_meds      float64
behavioral_avoidance           float64
behavioral_face_mask           float64
behavioral_wash_hands          float64
behavioral_large_gatherings    float64
behavioral_outside_home        float64
behavioral_touch_face          float64
doctor_recc_h1n1               float64
doctor_recc_seasonal           float64
chronic_med_condition          float64
child_under_6_months           float64
health_worker                  float64
health_insurance               float64
opinion_h1n1_vacc_effective    float64
opinion_h1n1_risk              float64
opinion_h1n1_sick_from_vacc    float64
opinion_seas_vacc_effective    float64
opinion_seas_risk              float64
opinion_seas_sick_from_vacc    float64
age_group                       object
education                       object
race                            object
sex                      

In [30]:
race.info()

<class 'pandas.core.series.Series'>
RangeIndex: 26707 entries, 0 to 26706
Series name: race
Non-Null Count  Dtype 
--------------  ----- 
26707 non-null  object
dtypes: object(1)
memory usage: 208.8+ KB


In [37]:
race_column = df['race'].value_counts()
race_column

race
White                21222
Black                 2118
Hispanic              1755
Other or Multiple     1612
Name: count, dtype: int64


In [39]:
sex_column = df['sex'].value_counts()
sex_column

sex
Female    15858
Male      10849
Name: count, dtype: int64

Summary of the dataset
From the datasets, there are four race groups; white, black, hispanic, and others. The majority was white. The female population was at higher than male at 15858 while 10849 for the male. 

Data Cleaning 

In [26]:
# checking for duplicates
df.duplicated().sum()

0

In [15]:
# check missing values
df.isnull().sum()

respondent_id                      0
h1n1_concern                      92
h1n1_knowledge                   116
behavioral_antiviral_meds         71
behavioral_avoidance             208
behavioral_face_mask              19
behavioral_wash_hands             42
behavioral_large_gatherings       87
behavioral_outside_home           82
behavioral_touch_face            128
doctor_recc_h1n1                2160
doctor_recc_seasonal            2160
chronic_med_condition            971
child_under_6_months             820
health_worker                    804
health_insurance               12274
opinion_h1n1_vacc_effective      391
opinion_h1n1_risk                388
opinion_h1n1_sick_from_vacc      395
opinion_seas_vacc_effective      462
opinion_seas_risk                514
opinion_seas_sick_from_vacc      537
age_group                          0
education                       1407
race                               0
sex                                0
income_poverty                  4423
m

0